app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -575,40 +576,89 @@ class HomeActivityTest {
     }
   }
 
+  @Config(qualifiers = "port")
   @Test
-  fun testHomeActivity_noTopicsStarted_displaysAllTopicCardsInPortrait() {
+  fun testHomeActivity_noTopicsStarted_mobilePortraitDisplaysTopicsIn2Columns() {
     // Only new users will have no progress for any topics.
-    profileTestHelper.loginToNewUser()
+    profileTestHelper.logIntoNewUser()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
+      hasGridColumnCount(expectedColumnCount = 2)
+
+      scrollToPosition(position = 3)
+      onView(withId(R.id.home_recycler_view)).check(
+        hasGridItemCount(spanCount = 1, position = 3)
+      )
+    }
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testHomeActivity_noTopicsStarted_mobileLandspaceDisplaysTopicsIn3Columns() {
+    // Only new users will have no progress for any topics.
+    profileTestHelper.logIntoNewUser()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      hasGridColumnCount(expectedColumnCount = 3)
+
+      scrollToPosition(position = 3)
+      onView(withId(R.id.home_recycler_view)).check(
+        hasGridItemCount(spanCount = 1, position = 3)
+      )
+    }
+  }
+
+  @Config(qualifiers = "sw600dp-port")
+  @Test
+  fun testHomeActivity_noTopicsStarted_tabletPortraitDisplaysTopicsIn3Columns() {
+    // Only new users will have no progress for any topics.
+    profileTestHelper.logIntoNewUser()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      hasGridColumnCount(expectedColumnCount = 3)
+
       scrollToPosition(position = 3)
       onView(withId(R.id.home_recycler_view)).check(
-        // The "All Topics" section currently should display all four test topics in two rows.
-        hasGridColumnCount(expectedColumnCount = 2)
+        hasGridItemCount(spanCount = 1, position = 3)
       )
     }
   }
 
+  @Config(qualifiers = "sw600dp-land")
   @Test
-  fun testHomeActivity_noTopicsStarted_displaysAllTopicCardsInLandscape() {
+  fun testHomeActivity_noTopicsStarted_tabletLandscapeDisplaysTopicsIn4Columns() {
     // Only new users will have no progress for any topics.
-    profileTestHelper.loginToNewUser()
+    profileTestHelper.logIntoNewUser()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
       onView(isRoot()).perform(orientationLandscape())
+      hasGridColumnCount(expectedColumnCount = 4)
```

Comment:
Her & ablove: does this actually fail? Doesn't this need to be run as part of an onView() setup?

Reply:
This part shouldn't fail, this is meant to be part of the test set-up. So since we're checking `hasGridItemCount(spanCount = 1, ...)` this only tells us that the item has a span of 1,. But having this additional `hasGridColumnCount(expectedColumnCount = 4)` ensures that if the time has a span count of 1, then the home fragment is going to be displaying the items in a row of 4. Does that make sense / is there a better way to be checking this?

Will add a comment for now to clarify this.

Reply:
Done

Reply:
Sorry, what I meant was: I don't think 'hasGridColumnCount' was _doing_ anything. The verify addresses my concern, though, since it can now actually fail.